src/faction_camp.cpp

```diff
@@ -1838,7 +1838,7 @@ void basecamp::job_assignment_ui()
                 selection--;
             }
         } else if( action == "CONFIRM" ) {
-            if( !stationed_npcs.empty() ) {
+            if( cur_npc ) {
```

Comment:
Forgot to mention in the original description: this fix is for an unrelated null pointer dereference warning that clang-tidy also gave me while working on this PR.